### PR TITLE
fix: DBTP-1454 Limit addons parameter to current environment

### PR DESCRIPTION
**Pull request by WillGibson**
Addresses https://uktrade.atlassian.net/browse/DBTP-1454.

Parameter before apply with updated `terraform-platform-modules`…

image

Parameter after apply with updated `terraform-platform-modules`…

image

### CODE REVIEW

File: extensions/locals.tf [hunk 1]
```diff
@@ -1,6 +1,16 @@
...
 
+  extensions_for_environment = {
+    for k, v in var.args.services :
+    k => merge(v, {
+      environments = {
+        for ek, ev in v["environments"] :
+        ek => ev if contains(["*", var.environment], ek)
+      }
+    })
+  }
```
Comment (WillGibson) [inline extensions/locals.tf:12]: Kudos to @acodeninja for this fix.

Comment (a-gleeson) [reply extensions/locals.tf:12]: should we have a small comment to describe what is happening here?

**Review by a-gleeson — APPROVED** [review]
LGTM!

File: extensions/locals.tf [hunk 2]
```diff
@@ -1,6 +1,16 @@
 locals {
   plans = yamldecode(file("${path.module}/plans.yml"))
 
+  extensions_for_environment = {
```
Comment (WillGibson) [inline extensions/locals.tf:4]:
```suggestion
  # So we don't hit a Parameter Store limit, filter environment config for extensions to just the defaults (`"*"`) and the current environment
  extensions_for_environment = {
```

**Pull request merged**